### PR TITLE
Mongo db

### PR DESCRIPTION
**Pull request by Anthony271828**
Implements the wrapper to properly interact with the database.

Implemented Wrapper:
- _Subjects_
- _StudyChannels_
- _GamingChannels_
- Generic wrapper for primitive purposes

Also implemented _CollectionEnum_ to provide all currently used database collections

### CODE REVIEW

**Review by maxwai — CHANGES_REQUESTED** [review]
see comments. I'm leaving it to @Anthony271828 that the methods themselves work

File: Mongo/gamingChannels.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+from discord.ext.commands import Bot
+
+from Mongo.mongocollection import MongoCollection, MongoDocument
+from discord import Member, TextChannel, VoiceChannel, User, Guild
+import datetime
+
+
+@dataclass
+class GamingChannel(MongoDocument):
```
[the remark this answers is not in the document]
Comment (Anthony271828) [reply Mongo/gamingChannels.py:14]: If frozen would be set to True, then the token could not be changed.

File: Mongo/studyChannels.py [hunk 2]
```diff
@@ -8,6 +9,9 @@
 from discord import Member, TextChannel, VoiceChannel, User, Guild
 import datetime
 
+from Mongo.primitivemongodata import PrimitiveMongoData
+from core.globalenum import CollectionEnum
+
 
 @dataclass
 class StudyChannel(MongoDocument):
```
[the remark this answers is not in the document]
Comment (Anthony271828) [reply Mongo/studyChannels.py:17]: If frozen would be set to True, then the token could not be changed, for example.

File: Mongo/gamingChannels.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            "chat": self.chat.id,
+            "voice": self.voice.id,
+            "token": self.token,
+        }
+
+
+# noinspection DuplicatedCode
+class GamingChannels(MongoCollection):
+    def __init__(self, bot: Bot):
+        super().__init__(os.environ["DB_NAME"], self.__class__.__name__)
```
[the remark this answers is not in the document]
Comment (Anthony271828) [reply Mongo/gamingChannels.py:34]: We could do a default, but I would still like to leave the option open.

File: Mongo/gamingChannels.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+        return GamingChannel(_id, owner, chat, voice, token)
+
+    async def insert_one(self,
+                         entry: tuple[
+                             Union[Member, User],
+                             TextChannel, VoiceChannel,
+                             int,
+                             Optional[datetime.datetime]]) -> GamingChannel:
+        owner, chat, voice, token, delete_at = entry
```
[the remark this answers is not in the document]
Comment (Anthony271828) [reply Mongo/gamingChannels.py:52]: This is to leave open the option to set a deadline beyond the default. This should allow for special long term chats if needed.

File: Mongo/studyChannels.py [hunk 5]
```diff
@@ -57,16 +56,18 @@ async def insert_one(self,
                              Optional[datetime.datetime]]) -> StudyChannel:
         owner, chat, voice, token, delete_at = entry
 
+        hours = await PrimitiveMongoData.find_configuration(CollectionEnum.ROLES_SETTINGS, "deleteAfter", "HOURS")
```
[the remark this answers is not in the document]
Comment (Anthony271828) [reply Mongo/studyChannels.py:58]: Will fix that!

File: Mongo/gamingChannels.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+        document = {
+            "owner": owner.id,
+            "chat": chat.id,
+            "voice": voice.id,
+            "token": token,
```
[the remark this answers is not in the document]
Comment (Anthony271828) [reply Mongo/gamingChannels.py:58]: Will also fix that!

File: Mongo/subjects.py [hunk 7]
```diff
@@ -29,7 +29,7 @@ def channel_id(self):
         return self.chat.id
```
[the remark this answers is not in the document]
Comment (Anthony271828) [reply Mongo/subjects.py:30]: Now that you mention it...

**Comment by Anthony271828** [comment]
I hope that your demands have been adequately implemented.

The reason for adding three more enums is to ensure a better distinction regarding the purpose of the strings. Even if the naming is crappy, I hope that the purpose is clear.

File: Mongo/gamingChannels.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (maxwai) [reply Mongo/gamingChannels.py:52]: Yes then de a default and leave the option open

File: Mongo/gamingChannels.py [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+        return {member for member in self.voice.members if not member.bot}
+
+    @property
+    def document(self) -> dict[str: typing.Any]:
+        return {
+            "_id": self._id,
+            "owner": self.owner.id,
+            "chat": self.chat.id,
+            "voice": self.voice.id,
+            "token": self.token,
```
Comment (maxwai) [inline Mongo/gamingChannels.py:46]: Forgot to add enum values here

**Review by maxwai — CHANGES_REQUESTED** [review]
forgot one thing then it is fine

**Comment by Anthony271828** [comment]
Your change request should be fully done.

**Review by maxwai — APPROVED** [review]

**Pull request merged**